cron/k8s/worker.yaml

```diff
@@ -0,0 +1,49 @@
+# Copyright 2021 Security Scorecard Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: scorecard-batch-worker
+spec:
+  replicas: 5
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: worker
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: worker
+    spec:
+      containers:
+      - name: worker
+        image: gcr.io/openssf/scorecard-batch-worker
+        imagePullPolicy: Always
+        env:
+        - name: GITHUB_AUTH_TOKEN
+          valueFrom:
+            secretKeyRef:
+              name: github
+              key: testToken
+        - name: USE_BLOB_CACHE
+          value: "true"
+        - name: BLOB_URL
+          value: "gs://ossf-scorecards-cache"
+        resources:
+          requests:
+            memory: 10Gi
+            cpu: 100
+          limits:
+            memory: 15Gi
+            cpu: 220
```

Comment:
that's quite beefy.. any reason why we need 220 CPUs machines, rather than more (smaller) machines? this workload seems like something that inherently better suits horizontal scaling instead.

Reply:
+1

Reply:
Yes they are, was hoping to get some feedback here on it. I simply did some rough back of the hand calculations and set them to pretty high numbers. Basically, each repo runs 14 threads in parallel and we are hoping to process maybe 10 repos in parallel in a single machine. I have assumed 1vCPU~=1 thread, so set the CPU requirements to ~150 CPUs.

What do you guys think? Whats a reasonable value to set here.

Reply:
These threads are mostly IO bound right? We can probably get by with 1 CPU core per repo (at most). We can start with a very small number of CPUs and monitor the CPU usage and adjust as needed. 220 seems a bit overkill to me.

Also, it seems more beneficial to run 10 separate workers rather than 10 repos in parallel. Are there benefits to running so many on the one worker?

Reply:
Nope, no benefit IMO. Like I said, I was just doing some rough calculations. I have reduced the CPU numbers to req:8 and limit:12. Should be a good starting point?

Reply:
In that case, honestly I think we can get back with (req:1, limit:1.5). Scorecards shouldn't be too CPU intensive. We can always monitor and increase as needed.

Reply:
Done.